tests/bulk_create/models.py

```diff
@@ -81,6 +93,9 @@ class NullableFields(models.Model):
     positive_small_integer_field = models.PositiveSmallIntegerField(null=True, default=4)
     small_integer_field = models.SmallIntegerField(null=True, default=5)
     time_field = models.TimeField(null=True, default=timezone.now)
+    auto_field = models.ForeignKey(AutoFieldModel, on_delete=models.CASCADE, null=True)
+    small_auto_field = models.ForeignKey(SmallAutoFieldModel, on_delete=models.CASCADE, null=True)
+    big_auto_field = models.ForeignKey(BigAutoFieldModel, on_delete=models.CASCADE, null=True)
```

Comment:
These are not a proper regression tests. ticket-31301 describes an issue with mixed types, i.e. empty and set foreign keys in a single `bulk_create()`, e.g.
```diff
diff --git a/tests/bulk_create/models.py b/tests/bulk_create/models.py
index 87203ac4f3..e97d106029 100644
--- a/tests/bulk_create/models.py
+++ b/tests/bulk_create/models.py
@@ -53,6 +53,7 @@ class Pizzeria(Restaurant):
 
 class State(models.Model):
     two_letter_code = models.CharField(max_length=2, primary_key=True)
+    country = models.ForeignKey(Country, on_delete=models.CASCADE, null=True)
 
 
 class TwoFields(models.Model):
diff --git a/tests/bulk_create/tests.py b/tests/bulk_create/tests.py
index 2fd9801d35..9b20acb15e 100644
--- a/tests/bulk_create/tests.py
+++ b/tests/bulk_create/tests.py
@@ -60,6 +60,15 @@ class BulkCreateTests(TestCase):
         ])
         self.assertEqual(Country.objects.count(), 4)
 
+    @skipUnlessDBFeature('has_bulk_insert')
+    def test_empty_set_foreign_key(self):
+        country = Country.objects.create(description='test')
+        State.objects.bulk_create([
+            State(two_letter_code='NY'),
+            State(two_letter_code='IL', country=country),
+        ])
+        self.assertEqual(State.objects.count(), 2)
+
     def test_multi_table_inheritance_unsupported(self):
         expected_message = "Can't bulk create a multi-table inherited model"
         with self.assertRaisesMessage(ValueError, expected_message):
```

Reply:
There are mixed types - in `test_bulk_insert_nullable_fields`, the fields by default are filled and while looping over every field, they are set to None 1 at a time

Reply:
Right.